src/main/scala/is/hail/utils/richUtils/RichIterableGenotype.scala

```diff
@@ -0,0 +1,8 @@
+package is.hail.utils.richUtils
+
+import is.hail.utils.{IntIterator, HardCallIterator}
+import is.hail.variant.Genotype
+
+class RichIterableGenotype(it: Iterable[Genotype]) {
+  def hardCallIterator: HardCallIterator = new HardCallIterator(it.iterator)
```

Comment:
I thought the implicit would only be called if gs was not a GenotypeStream, since that's the more specialized class on which hardCallIterator is defined.

Is this what you have in mind?
```
class RichIterableGenotype(it: Iterable[Genotype]) {
  def hardCallIterator: IntIterator = it match {
    case gs: GenotypeStream => gs.hardCallIterator
    case _ =>
      val iter = it.iterator
      class HardCallIterator extends IntIterator {
        override def hasNext: Boolean = iter.hasNext
        override def nextInt(): Int = iter.next().unboxedGT
      }
      new HardCallIterator
  }
}
```

Or this?

```
class RichIterableGenotype(it: Iterable[Genotype]) {
  def hardCallIterator: IntIterator = it match {
    case gs: GenotypeStream => gs.hardCallIterator
    case _ =>
      val iter = it.map(_.unboxedGT).iterator
      class HardCallIterator extends IntIterator {
        override def hasNext: Boolean = iter.hasNext
        override def nextInt(): Int = iter.next()
      }
      new HardCallIterator
  }
}
```

 Or something else entirely?